mage_integrations/mage_integrations/destinations/oracledb/__init__.py

```diff
@@ -0,0 +1,216 @@
+    ) -> List[str]:
+        columns = list(schema['properties'].keys())
+        insert_columns, insert_values = build_insert_command(
+            column_type_mapping=column_type_mapping(
+                schema,
+                convert_column_type,
+                lambda item_type_converted: 'LONGTEXT',
+            ),
+            columns=columns,
+            records=records,
+            string_parse_func=lambda x, y: x.replace("'", "''").replace('\\', '\\\\')
+            if COLUMN_TYPE_OBJECT == y['type'] else x,
+        )
+        insert_columns = ', '.join([clean_column_name(col) for col in insert_columns])
+        insert_into = f'INTO {table_name} ({insert_columns})'
+        commands = []
+        columns_cleaned = [clean_column_name(col) for col in columns]
+
+        for insert_value in insert_values:
+            if unique_constraints and unique_conflict_method:
+                if UNIQUE_CONFLICT_METHOD_UPDATE == unique_conflict_method:
+                    # Build update command
+                    insert_value_without_left_parenthesis = re.sub(r"^\(", "", insert_value)
+                    insert_value_strip = re.sub(r"\)$", "", insert_value_without_left_parenthesis)
+                    updated_values = insert_value_strip.split(',')
+                    updated_command = ', '.join([f'{col} = {updated_values[idx].strip()}'
+                                                for idx, col in enumerate(columns_cleaned)])
+                    update_command_constraint = ""
+                    for idx, column in enumerate(columns):
+                        if column in unique_constraints:
+                            if len(update_command_constraint) == 0:
+                                update_command_constraint += \
+                                    f'{clean_column_name(column)} = {updated_values[idx]}'
+                            else:
+                                update_command_constraint += \
+                                    f'AND {clean_column_name(column)} = {updated_values[idx]}'
+
+                    insert_command = f'''
+BEGIN
+    INSERT {insert_into} VALUES {insert_value};
+EXCEPTION
+    WHEN DUP_VAL_ON_INDEX
+    THEN
+        UPDATE {table_name}
+        SET {updated_command}
+        WHERE {update_command_constraint};
+    WHEN OTHERS THEN
+        RAISE;
+END;
+'''
+            else:
+                # When there is no unique_constraints
+                insert_command = f'INSERT {insert_into} VALUES {insert_value}'
+            commands.append(insert_command)
+
+        return commands
+
+    def does_table_exist(
+        self,
+        schema_name: str,
+        table_name: str,
+        database_name: str = None,
+    ) -> bool:
+        connection = self.build_connection().build_connection()
+        cursor = connection.cursor()
+        cursor.execute(
+            f'SELECT COUNT(*) FROM user_tables WHERE table_name = \'{table_name.upper()}\'')
+        (number_of_rows,) = cursor.fetchone()
```

Comment:
you can just do `number_of_rows = cursor.fetchone()[0]`

Reply:
Updated